proposals/0000-set-execution-hint.md

```diff
@@ -0,0 +1,79 @@
+---
+Title: Add SetExecutionHints instruction
+Number: 0
+Status: Draft
+Version: 5
+Authors:
+ - Francisco Aguirre
+Created: 2024-05-23
+Impact: Medium
+Requires: 37
+Replaces:
+---
+
+## Summary
+
+RFC#37 introduces a `SetAssetClaimer` instruction.
+This idea of instructing the XCVM to change some implementation-specific behavior is useful.
+In order to generalize this mechanism, this RFC introduces a new instruction `SetExecutionHints`
+and makes the `SetAssetClaimer` be just one of many possible execution hints.
+
+## Motivation
+
+There is a need for specifying how certain implementation-specific things should behave.
+Things like who can claim the assets or what can be done instead of trapping assets.
+
+## Specification
+
+A new instruction, `SetExecutionHints`, will be added.
+This instruction will take a single parameter of type `ExecutionHint`, an enumeration.
+The first variant for this enum is `AssetClaimer`, which allows to specify a location that should be able to claim trapped assets,
+as specified in RFC#37.
+This means the instruction `SetAssetClaimer` would also be removed, in favor of this.
+
+In Rust, the new definitions would look as follows:
+
+```rust
+enum Instruction {
+  // ...snip...
+  SetExecutionHints(BoundedVec<ExecutionHint, NumVariants>),
+  // ...snip...
+}
+
+enum ExecutionHint {
+  AssetClaimer(Location),
+  // more can be added
+}
+
+type NumVariants = /* Number of variants of the `ExecutionHint` enum */;
+```
+
+## Security considerations
+
+`ExecutionHint`s are specified on a per-message basis, so they have to be specified at the beginning of a message.
+If they were to be specified at the end, hints like `AssetClaimer` would be useless if an error occurs beforehand and assets get trapped before ever reaching the hint.
+
+The instruction takes a bounded vector of hints so as to not 
```

Comment:
It seems like an unfinished sentence.

I assume the message is that the vector in the `SetExecutionHints` may contain a maximum of `NumVariants` items. The vector is guaranteed to be small so that it won't significantly increase the size of XCM messages.

Reply:
Don't know why it was left unfinished, corrected